mkt/site/tests/test_views.py

```diff
@@ -66,6 +66,13 @@ def test_404_api(self):
         res = self.client.get('/api/this-should-never-work/')
         eq_(res.status_code, 404)
         eq_(res.content, '')
+        self.assertCORS(res, 'get')
+
+    def test_404_api_debug(self):
+        with self.settings(DEBUG=True):
+            res = self.client.get('/api/this-should-never-work/')
+            eq_(res.status_code, 404)
+            self.assertCORS(res, 'get')
```

Comment:
I've noticed some subtle bugs where options requests are not handled correctly and that's the first thing the browser does before any cors logic. So it would be good to add this in all tests like:

    res = self.client.options('/api/this-should-never-work/')
    self.assertCORS(res, 'options')